helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -296,27 +332,21 @@ public void close() {
       Set<String> activeNodes, final CurrentStateOutput currentStateOutput)
       throws HelixRebalanceException {
     getChangeDetector().updateSnapshots(clusterData);
-    // Get all the changed items' information
-    Map<HelixConstants.ChangeType, Set<String>> clusterChanges =
+    // Get all the changed items' information. Filter for the items that have content changed.
+    final Map<HelixConstants.ChangeType, Set<String>> clusterChanges =
```

Comment:
can you parallel stream this?

Reply:
As Yi pointed out, parallelStream is costly. So I tested multiple places. The conclusion is that only high-level parallelism is worthwhile. Doing it too much slow things down.

For this one, the change types are not a long list. So the gain won't beat the thread managing overhead.